zkevm-circuits/src/table/keccak_table.rs

```diff
@@ -144,3 +152,144 @@ impl KeccakTable {
         ]
     }
 }
+
+/// Keccak Table, used to verify keccak hashing from hi/lo input.
+#[derive(Clone, Debug)]
+pub struct KeccakTable2 {
+    /// True when the row is enabled
+    pub is_enabled: Column<Advice>,
+    /// Byte array input as `RLC(reversed(input))`
+    pub input_rlc: Column<Advice>, // RLC of input bytes
+    /// Byte array input length
+    pub input_len: Column<Advice>,
+    /// hash high 16 bytes
+    pub output_hi: Column<Advice>,
+    /// hash low 16 bytes
+    pub output_lo: Column<Advice>,
+}
+
+impl<F: Field> LookupTable<F> for KeccakTable2 {
+    fn columns(&self) -> Vec<Column<Any>> {
+        vec![
+            self.is_enabled.into(),
+            self.input_rlc.into(),
+            self.input_len.into(),
+            self.output_hi.into(),
+            self.output_lo.into(),
+        ]
+    }
+
+    fn annotations(&self) -> Vec<String> {
+        vec![
+            String::from("is_enabled"),
+            String::from("input_rlc"),
+            String::from("input_len"),
+            String::from("output_hi"),
+            String::from("output_lo"),
+        ]
+    }
+}
+
+impl KeccakTable2 {
+    /// Construct a new KeccakTable
+    pub fn construct<F: Field>(meta: &mut ConstraintSystem<F>) -> Self {
+        Self {
+            is_enabled: meta.advice_column(),
+            input_rlc: meta.advice_column_in(SecondPhase),
+            input_len: meta.advice_column(),
+            output_hi: meta.advice_column(),
+            output_lo: meta.advice_column(),
+        }
+    }
+
+    /// Generate the keccak table assignments from a byte array input.
+    pub fn assignments<F: Field>(
+        input: &[u8],
+        challenges: &Challenges<Value<F>>,
+    ) -> Vec<[Value<F>; 5]> {
+        const BYTE_POW_BASE: u64 = 1 << 8;
+        let input_rlc = challenges
+            .keccak_input()
+            .map(|challenge| rlc::value(input.iter().rev(), challenge));
+        let input_len = F::from(input.len() as u64);
+        let mut keccak = Keccak::default();
+        keccak.update(input);
+        let output = keccak.digest();
+
+        let output_hi = output.iter().take(16).fold(F::ZERO, |acc, byte| {
+            acc * F::from(BYTE_POW_BASE) + F::from(*byte as u64)
+        });
+
+        let output_lo = output.iter().skip(16).fold(F::ZERO, |acc, byte| {
+            acc * F::from(BYTE_POW_BASE) + F::from(*byte as u64)
+        });
```

Comment:
Could https://github.com/taikoxyz/zkevm-circuits/pull/129/files#diff-55f686c17b36ef6bb92c6d15364561ab33f6f2c7ed2df3334f185a10cf49623bR396-R405 be reused here?